ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -40,25 +63,169 @@ private ListQueriesExecutor() { }
   ) {
     final ListQueries listQueries = statement.getStatement();
     if (listQueries.getShowExtended()) {
-      return Optional.of(new QueryDescriptionList(
-          statement.getStatementText(),
-          executionContext.getPersistentQueries().stream()
-              .map(QueryDescriptionFactory::forQueryMetadata)
-              .collect(Collectors.toList())));
+      return executeExtended(
+          statement,
+          sessionProperties,
+          executionContext,
+          serviceContext);
+    }
+
+    final List<RunningQuery> runningQueries = executionContext.getPersistentQueries()
+        .stream()
+        .map(q -> {
+          final KafkaStreamsStateCount kafkaStreamsStateCount = new KafkaStreamsStateCount();
+          kafkaStreamsStateCount.updateStateCount(q.getState(), 1);
+          return new RunningQuery(
+              q.getStatementString(),
+              ImmutableSet.of(q.getSinkName().text()),
+              ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
+              q.getQueryId(),
+              kafkaStreamsStateCount
+          );
+        }).collect(Collectors.toList());
+
+    final Map<String, RunningQuery> queryToRunningQuery = runningQueries.stream().collect(
+        Collectors.toMap(
+            query -> query.getId().toString(),
+            query -> query));
+
+    if (!sessionProperties.getInternalRequest()) {
+      final Set<HostInfo> remoteHosts =
+          DiscoverRemoteHostsUtil.getRemoteHosts(
+              executionContext.getPersistentQueries(),
+              sessionProperties.getKsqlHostInfo());
+      
+      if (remoteHosts.size() != 0) {
+        final ExecutorService executorService = Executors.newFixedThreadPool(remoteHosts.size());
+        final List<Future<List<RunningQuery>>> futureRunningQueries = new ArrayList<>();
+        for (HostInfo host : remoteHosts) {
+          futureRunningQueries.add(executorService.submit(() -> {
+            final KsqlEntityList response = serviceContext.getKsqlClient()
+                .makeKsqlRequestWithRequestProperties(
+                    ServerUtil.buildRemoteUri(
+                        sessionProperties.getLocalUrl(),
+                        host.host(),
+                        host.port()
+                    ),
+                    statement.getStatementText(),
+                    Collections.singletonMap(KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST, true))
+                .getResponse();
+            return ((Queries) response.get(0)).getQueries();
+          }));
+        }
+
+        final ConcurrentLinkedQueue<RunningQuery> remoteRunningQueries =
+                new ConcurrentLinkedQueue<>();
+        futureRunningQueries.forEach(future -> {
+          try {
+            remoteRunningQueries.addAll(future.get());
+          } catch (final Exception e) {
+            // If the future fails from a server, that result won't be included in the output
```

Comment:
I'm not sure how that would exactly fit in with the current outputs that's proposed. Would it be fine to just log the failure for now and I'll file a issue for improving the response to indicate unresponsive hosts?

Reply:
If you're planning on do that as the next task, then sure, by all means get this one merged and then enhance to include unresponsive hosts.

If you mean just raise it as a task and then it gets lost in the backlog... never to see the light of day again... ;).   Then I'd say no.  

This piece of work you're doing, (which is awesome!), would be incomplete IMHO if the user wasn't informed that the results were incomplete.  One of the main reasons they may be looking into the query status on the different hosts is because something is wrong and they're investigating.  Hence we should be providing them with the info they need, not hiding errors from them.

The current output has two modes, right?

Normal outputs something like `status: RUNNING:1, CREATED: 2`.  So why not include a `UNRESPONSIVE: 2` in that list?

Extended output goes a step further and outputs the state on each host, e.g. `Host Query Status    : {Host1=RUNNING, Host2=RUNNING}`. This too could easily be extended to include `Host3=UNRESPONSIVE`.